geyser-plugin-manager/src/geyser_plugin_manager.rs

```diff
@@ -29,21 +38,115 @@ impl GeyserPluginManager {
         libpath: &str,
         config_file: &str,
     ) -> Result<(), Box<dyn Error>> {
-        type PluginConstructor = unsafe fn() -> *mut dyn GeyserPlugin;
+        // Load the dynamic library and reconstruct the Box<dyn GeyserPlugin>
         let lib = Library::new(libpath)?;
         let constructor: Symbol<PluginConstructor> = lib.get(b"_create_plugin")?;
         let plugin_raw = constructor();
-        let mut plugin = Box::from_raw(plugin_raw);
+        let mut plugin: Box<dyn GeyserPlugin> = Box::from_raw(plugin_raw);
```

Comment:
Is this change needed? Avoid unnecessary changes if not

Reply:
Not needed. I'll remove it